grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/fragment/IdFragment.java

```diff
@@ -38,8 +40,24 @@
     }
 
     @Override
-    public void applyTraversal(GraphTraversal<Vertex, Vertex> traversal, GraknGraph graph) {
-        traversal.has(Schema.VertexProperty.ID.name(), id.getValue());
+    public void applyTraversal(GraphTraversal<? extends Element, ? extends Element> traversal, GraknGraph graph) {
+        if (operatesOnEdge()) {
+            // Handle both edges and vertices
+            traversal.or(
+                    edgeTraversal(),
+                    vertexTraversal(__.identity())
+            );
+        } else {
+            vertexTraversal(traversal);
+        }
+    }
+
+    private <S, E> GraphTraversal<S, E> vertexTraversal(GraphTraversal<S, E> traversal) {
+        return traversal.has(Schema.VertexProperty.ID.name(), id.getValue());
```

Comment:
Maybe worth noting here why the separation is needed. Why can't vertices and edges be looked up by `hasId(xxxxx)` ?